dspy/clients/lm.py

```diff
@@ -246,6 +246,10 @@ def copy(self, **kwargs):
                 new_instance.kwargs[key] = value
 
         return new_instance
+    
+    def dump_state(self):
+        # TODO: callbacks cannot be saved. We should consider how to save callbacks.
```

Comment:
good catch! actually callbacks doesn't belong to the state, it's part of runtime config, we don't need to save it with this `dump_state`